src/main/java/functions/VisitorFunction.kt

```diff
@@ -1,18 +1,22 @@
 package functions
 
 import ast.ASTNode
+import ast.BreadthClosure
 import ast.Child
 import ast.Node
 import ast.Sibling
 import tree.TargetTreeNode
 
-typealias VisitorFunction = (TargetTreeNode?) -> List<TargetTreeNode>
+data class VisitorFunctionResponse(val matches: List<TargetTreeNode> = listOf(), val hasMatch: Boolean = false)
```

Comment:
This is the cornerstone of this change and a much needed data structure to transfer data among functions. Well done!

Reply:
![](https://media.giphy.com/media/l0Ex87RyD4YUvBKPC/giphy.gif)